pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go

```diff
@@ -112,16 +116,84 @@ type DistributionInfo struct {
+// The original definition is from https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go
+type OCPVersionRelease struct {
+	// version is a semantic versioning identifying the update version. When this
+	// field is part of spec, version is optional if image is specified.
+	Version string `json:"version"`
+
+	// image is a container image location that contains the update. When this
+	// field is part of spec, image is optional if version is specified and the
+	// availableUpdates field contains a matching version.
+	Image string `json:"image"`
+
+	// url contains information about this release. This URL is set by
+	// the 'url' metadata property on a release or the metadata returned by
+	// the update API and should be displayed as a link in user
+	// interfaces. The URL field may not be set for test or nightly
+	// releases.
+	URL string `json:"url,omitempty"`
+
+	// channels is the set of Cincinnati channels to which the release
+	// currently belongs.
+	Channels []string `json:"channels,omitempty"`
+}
+
+// OCPVersionUpdateHistory is a single attempted update to the cluster.
+// the original definition is from https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go
+type OCPVersionUpdateHistory struct {
+	// state reflects whether the update was fully applied. The Partial state
+	// indicates the update is not fully applied, while the Completed state
+	// indicates the update was successfully rolled out at least once (all
+	// parts of the update successfully applied).
+	State string `json:"state"`
+
+	// version is a semantic versioning identifying the update version. If the
+	// requested image does not define a version, or if a failure occurs
+	// retrieving the image, this value may be empty.
+	Version string `json:"version"`
+
+	// image is a container image location that contains the update. This value
+	// is always populated.
+	Image string `json:"image"`
+
+	// verified indicates whether the provided update was properly verified
+	// before it was installed. If this is false the cluster may not be trusted.
+	Verified bool `json:"verified"`
+}
+
 // OCPDistributionInfo defines the distribution information of OCP managed cluster
 type OCPDistributionInfo struct {
-	// Version is the distribution version of OCP
+	// Version is the current version of the OCP cluster.
+	// Deprecated in release 2.3. Use clusterClaim version.openshift.io instead.
 	Version string `json:"version,omitempty"`
-	// availableUpdates contains the list of update versions that are appropriate for the manage cluster.
+
+	// AvailableUpdates contains the list of update versions that are appropriate for the manage cluster.
+	// Deprecated in release 2.3. Use VersionAvailableUpdates instead.
 	AvailableUpdates []string `json:"availableUpdates,omitempty"`
-	// Desiredversion represents the desired version of the ocp cluster
+
+	// DesiredVersion is the version that the cluster is reconciling towards
```

Comment:
also deprecate this?

Reply:
need add the whole fileds of [Desired](https://github.com/openshift/api/blob/a3c8fa53e01bc4442b687d3aca1571692f7676ce/config/v1/types_cluster_version.go#L87 )  to the clusterinfo if we deprecate this. I think `DesiredVersion` is enough.